### PR TITLE
add pull requests good practices

### PR DESCRIPTION
**Pull request by kaiomagalhaes**

### CODE REVIEW

File: engineering/dev_best_practices/project-structure/pull_requests.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+To do that, we will be creating an intermediary branch. The goal is to create a branch from the default branch and leave it clean. You will then create the PRs with your changes pointing to this clean branch instead of master. When all your PRs to this branch are merged, you can just create a PR from this branch to master. Example:
+
+1. create a base branch from the repository's default branch (usually develop or master), let's call it `refactor/my-cool-refactor`
+2. Work normally and make sure to create a branch with less than 15 files changed. It is okay if you have breaking changes, let's call it `refactor/replacing-this-with-that`
+3. Now you create a PR from `refactor/replacing-this-with-that` to `refactor/my-cool-refactor`
+4. You can then create all the PRs for this big change pointing to `refactor/my-cool-refactor`
+5. After all the PRs to ``refactor/my-cool-refactor` are done, you create a PR from `refactor/my-cool-refactor` to the default branch (usually develop or master)
```
Comment (alesmit) [inline engineering/dev_best_practices/project-structure/pull_requests.md]: Fix backtick typo

```suggestion
5. After all the PRs to `refactor/my-cool-refactor` are done, you create a PR from `refactor/my-cool-refactor` to the default branch (usually develop or master)
```

**Comment by alesmit** [comment]
@neonima I think the reason why we keep the number of files changed below 15 is that we don't want the reviewer to spend too much time and effort on it. Since it's not very likely to review generated files (we'd rather spend time reviewing the source files where those are generated from), I guess it wouldn't be a problem to use the standard approach, so collecting them in a single PR.

It'll be definitely a good point to add to the document before we merge this PR, but it would be great to hear other thoughts on it before.

**Comment by kaiomagalhaes** [comment]
@alesmit @neonima in which scenario would we have an amount of generated files that would make the 15 files rule obsolete?

**Comment by neonima** [comment]
@kaiomagalhaes this occurs often when we are dealing with proto based services. We can have small changes but since we are modifying the proto definition the amount of the generated files (language specific files, API documentation, postman files, OpenAPI, insomnia, Swagger, etc...) can exceeds 15 files even if the change is minimal

**Pull request merged**